### PR TITLE
Implement ConstraintMatrix::get_lines()

### PR DESCRIPTION
**Pull request by tamiko**
This function returns a range object of line iterators. This is particularly
useful to quickly iterate over constraints via

for(const auto &line : constraint_matrix.get_lines())
  // ...

In preparation of some cleanup of constrained_linear_operator().

In reference to #4704

### CODE REVIEW

**Comment by tamiko** [comment]
I wil let bob check this pull request.

File: include/deal.II/lac/constraint_matrix.h [hunk 1]
```diff
@@ -1120,6 +1122,90 @@ class ConstraintMatrix : public Subscriptor
...
+  typedef boost::iterator_range<const_iterator> LineRange;
+
+
+  /**
+   * Return a range object containing (const) iterators to all line entries
+   * stored in the ConstraintMatrix. Such a range is useful to initialize
+   * range-based for loops as supported by C++11.
+   *
+   * @return A range object for the half open range <code>[this->begin(),
+   * this->end())</code> of line entries
```
Comment (bangerth) [inline include/deal.II/lac/constraint_matrix.h]: add a period.

File: include/deal.II/lac/constraint_matrix.h [hunk 2]
```diff
@@ -1120,6 +1122,90 @@ class ConstraintMatrix : public Subscriptor
...
+   */
+  const LineRange get_lines() const
+  {
+    return boost::make_iterator_range(lines.begin(), lines.end());
```
Comment (bangerth) [inline include/deal.II/lac/constraint_matrix.h]: please move the function out of line into the .cc file

File: tests/lac/constraints_get_lines.cc [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+
+// test: set inhomogeneous constraints and indirectly apply those to other
+// constrained nodes.
```
Comment (bangerth) [inline tests/lac/constraints_get_lines.cc]: update comment?

**Comment by tamiko** [comment]
No regressions: https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&buildid=9810

**Comment by tamiko** [comment]
@bangerth Updated. I will run the testsuite again.

**Comment by tamiko** [comment]
Passes the testsuite.

**Pull request merged**